### PR TITLE
Fix ``re`` brain on ``3.11``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Not sure how we can do this easily with the CI. It's good to add it know to make sure the coverage stays as expected.
Only 25 failing tests left to go.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

### CODE REVIEW

File: astroid/brain/brain_re.py [hunk 1]
```diff
@@ -34,6 +35,30 @@ def _re_transform():
...
     DEBUG = sre_compile.SRE_FLAG_DEBUG
+    """
+        )
+    # On 3.11+ all flags come from re._compiler
+    return parse(
+        """
+    import re._compiler as _compiler
+    NOFLAG = 0
+    ASCII = _compiler.SRE_FLAG_ASCII
+    IGNORECASE = _compiler.SRE_FLAG_IGNORECASE
```
Comment (jacobtylerwalls) [inline astroid/brain/brain_re.py:46]: Is there a way to do this without reiterating the whole thing? Like
```python
if PY311_PLUS:
    import_compiler = "import re._compiler as _compiler"
else:
    import_compiler = "import sre_compiler as _compiler"
return parse(f"""
{import_compiler}
ASCII = _compiler.SRE_FLAG_ASCII
...
```
...unless there a functional impact from doing so?

Comment (DanielNoord) [reply astroid/brain/brain_re.py:46]: Probably. Let me have a look!

File: astroid/brain/brain_re.py [hunk 2]
```diff
@@ -7,33 +7,39 @@
...
-    # Since Python 3.6 there is the RegexFlag enum
-    # where every entry will be exposed via updating globals()
+def _re_transform() -> nodes.Module:
+    # The RegexFlag enum exposes all its entries by updating globals()
+    # On 3.11+ all flags come from re._compiler
+    if PY311_PLUS:
+        import_compiler = "import re._compiler as _compiler"
+    # In 3.6-3.10 all flags come from sre_compile
+    else:
+        import_compiler = "import sre_compiler as _compiler"
```
Comment (jacobtylerwalls) [inline astroid/brain/brain_re.py:21]: Oops.
```suggestion
        import_compiler = "import sre_compile as _compiler"
```

**Review by jacobtylerwalls — APPROVED** [review]
Great! So is the plan to let the pipeline fail until `numpy` can be built on 3.11?

**Comment by DanielNoord** [comment]
> Great! So is the plan to let the pipeline fail until `numpy` can be built on 3.11?

I tested locally without `numpy` installed, which passes the tests. Should we do that for now? Pin `numpy` to not install on `3.11`?

**Comment by jacobtylerwalls** [comment]
As long as we put it on a todo somewhere to turn it back on!

**Comment by DanielNoord** [comment]
Extracted CI changes into #1517 to make the diff a bit cleaner.

**Review by cdce8p — APPROVED** [review]
Thanks @DanielNoord!

**Pull request merged**